### PR TITLE
#111 Make use of custom adapter for recycler view's 

### PR DESCRIPTION
**Pull request by victorvicari**
enabled the android data binding on gradle

to increase code readability I've created three new classes that are responsible for all the recycerview's adapter.

added a binding adapter file to deal with some custom data bindings (with extension function). All the adapters now extends from single layout adapter. the layout it's the first argument passed for the adapter, and it's directly pushed to above layers. replaced the fragment context for actions with kotlin lambdas, to make more cleaner implementation. In some cases(two, to be more precise), I've created a new include layout because the same behavior was used in two different adapters.

### CODE REVIEW

**Comment by ujjwalagrawal17** [comment]
@victorvicari thanks for the PR, We are trying to implement #80 so that all our methods and classes have the documentation. It will be nice if you can add the documentation of the new methods, classes. Also, we will try to make sure that we will merge the PRs only after they pass Codacy PR quality review, Travis Build. Can you look into the reviews by Codacy?

File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/common/BindingAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+/**
+ * Use Kotlin [ExtensionFunctionType] to load format and show the date with [BindingAdapter]
+ ***/
+@SuppressLint("SimpleDateFormat")
+@BindingAdapter("dateToString")
+fun TextView.setDate(date: Date?) {
+    val dateFormat = SimpleDateFormat(this.context.getString(R.string.time_format_12_hours))
+    date?.let { this.text = dateFormat.format(it).toString() }
+}
```
Comment (ujjwalagrawal17) [inline app/src/main/java/org/wikiedufoundation/wikiedudashboard/common/BindingAdapter.kt]: Can we move these 2 functions in ViewUtils.kt file. We already have other extension functions there and we will only need 1 file to maintain.

File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/adapters/CourseUploadsRecyclerAdapter.kt [hunk 2]
```diff
@@ -1,70 +1,25 @@
...
 import org.wikiedufoundation.wikiedudashboard.ui.coursedetail.uploads.data.CourseUpload
 import org.wikiedufoundation.wikiedudashboard.ui.coursedetail.uploads.data.CourseUploadList
-import org.wikiedufoundation.wikiedudashboard.ui.coursedetail.uploads.view.CourseUploadsFragment
-import java.util.*
 
 /**
  * RecyclerView adapter for courses list data
- * @property courseUploadsFragment primary constructor property
+ * @param layoutId item layout id
+ * @param onClickListener using a high order function to implement o item click through a lambda
```
Comment (ujjwalagrawal17) [inline app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/adapters/CourseUploadsRecyclerAdapter.kt]: o click-> on click

File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/adapters/MyDashboardRecyclerAdapter.kt [hunk 3]
```diff
@@ -1,55 +1,24 @@
...
 import org.wikiedufoundation.wikiedudashboard.ui.dashboard.data.CourseListData
-import org.wikiedufoundation.wikiedudashboard.ui.dashboard.view.MyDashboardFragment
-import java.util.*
 
 /**
- * RecyclerView adapter for my dashboard
- * @constructor secondary constructor declaring [MyDashboardFragment] variable
+ * RecyclerView adapter for my dasboard list data
+ * @param layoutId item layout id
+ * @param onClickListener using a high order function to implement o item click through a lambda
```
Comment (ujjwalagrawal17) [inline app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/adapters/MyDashboardRecyclerAdapter.kt]: o -> on

**Comment by ujjwalagrawal17** [comment]
@victorvicari can you resolve the merge conflicts. We will try to merge this PR after that.

File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/common/BindingAdapter.kt (hunk 1 above)
Comment (victorvicari) [reply app/src/main/java/org/wikiedufoundation/wikiedudashboard/common/BindingAdapter.kt]: It's possible to move, of course. Although these are extension functions with binding purpose, only for layouts. Are you sure about this changes?

**Comment by ujjwalagrawal17** [comment]
Yes.

**Pull request merged**